src/util/typeVarinfoMap.ml

```diff
@@ -0,0 +1,43 @@
+open GoblintCil
+
+module TypeVarinfoMap : RichVarinfo.BiVarinfoMap.S with type t = CilType.Typ.t =
+struct
+
+
+  module Typ = struct
+    include CilType.Typ
+    let name_and_type_varinfo typ =
+      let name = CilType.Typ.show typ in
+      name, typ
+
+    let describe_varinfo v ts = ""
+  end
+
+  module TypeMap = RichVarinfo.BiVarinfoMap.Make(Typ)
+  include TypeMap
+  let to_varinfo (t: t) =
+    (* Invariant: Only sanitized types are inserted into TypeMap *)
+    let sanitize_type (t: Cil.typ) =
+      let t = unrollType t in
```

Comment:
Not sure what sanitized means here, but maybe it should be `unrollTypeDeep` then as well?